Projeto de análise descritiva, sobre o preço da gasolina comum no ano de 2022, nos postos Automoto - Automóveis do Amapá, com base em 12 conjuntos de dados, disponível na "[Série Histórica de Preços de Combustíveis](https://dados.gov.br/dados/conjuntos-dados/serie-historica-de-precos-de-combustiveis-por-revenda)", do Portal de Dados Abertos do governo.

# IDENTIFICAÇÃO DA POPULAÇÃO DO ESTUDO E DAS VARIÁVEIS DE INTERESSE
A população de estudo será dada com base nos dados dos postos Automoto no estado do Amapá.

## Manipulação de Dados
A biblioteca para manipulação de dados a ser utilizada neste projeto será o pandas.

### Importação do Pandas e leitura dos dataframes

In [ ]:
import pandas as pd

Serão utilizados 12 conjuntos de dados para o estudo, que serão concatenados em um único dataframe posteriormente.

In [ ]:
df_jan = pd.read_csv("/content/drive/MyDrive/estudo_etanol_gasolina_comum2022/precos-gasolina-etanol-01.csv",
                 sep=';')
df_jan.head()

Regiao - Sigla Estado - Sigla   Municipio                    Revenda  \
0              N             AC  RIO BRANCO  AUTO POSTO AMAPA - EIRELI   
1              N             AC  RIO BRANCO  AUTO POSTO AMAPA - EIRELI   
2              N             AC  RIO BRANCO  AUTO POSTO AMAPA - EIRELI   
3              N             AC  RIO BRANCO     AUTO POSTO ACAUAN LTDA   
4              N             AC  RIO BRANCO     AUTO POSTO ACAUAN LTDA   

       CNPJ da Revenda       Nome da Rua Numero Rua Complemento     Bairro  \
0   00.529.581/0001-53  VIA CHICO MENDES       3570         NaN      AREAL   
1   00.529.581/0001-53  VIA CHICO MENDES       3570         NaN      AREAL   
2   00.529.581/0001-53  VIA CHICO MENDES       3570         NaN      AREAL   
3   00.524.999/0001-78     RODOVIA AC 40       1800         NaN  VILA ACRE   
4   00.524.999/0001-78     RODOVIA AC 40       1800         NaN  VILA ACRE   

         Cep             Produto Data da Coleta Valor de Venda  \
0  69906-119            GASOLINA     03/01/2022           6,99   
1  69906-119              ETANOL     03/01/2022           5,99   
2  69906-119  GASOLINA ADITIVADA     03/01/2022           7,05   
3  69902-450            GASOLINA     03/01/2022           6,99   
4  69902-450              ETANOL     03/01/2022           6,12   

   Valor de Compra Unidade de Medida       Bandeira  
0              NaN        R$ / litro  VIBRA ENERGIA  
1              NaN        R$ / litro  VIBRA ENERGIA  
2              NaN        R$ / litro  VIBRA ENERGIA  
3              NaN        R$ / litro  VIBRA ENERGIA  
4              NaN        R$ / litro  VIBRA ENERGIA

In [ ]:
df_fev = pd.read_csv("/content/drive/MyDrive/estudo_etanol_gasolina_comum2022/precos-gasolina-etanol-02.csv",
                 sep=';')
df_mar = pd.read_csv("/content/drive/MyDrive/estudo_etanol_gasolina_comum2022/precos-gasolina-etanol-03.csv",
                 sep=';')
df_abr = pd.read_csv("/content/drive/MyDrive/estudo_etanol_gasolina_comum2022/precos-gasolina-etanol-04.csv",
                 sep=';')
df_mai = pd.read_csv("/content/drive/MyDrive/estudo_etanol_gasolina_comum2022/precos-gasolina-etanol-05.csv",
                 sep=';')
df_jun = pd.read_csv("/content/drive/MyDrive/estudo_etanol_gasolina_comum2022/precos-gasolina-etanol-06.csv",
                 sep=';')
df_jul = pd.read_csv("/content/drive/MyDrive/estudo_etanol_gasolina_comum2022/precos-gasolina-etanol-07.csv",
                 sep=';')
df_ago = pd.read_csv("/content/drive/MyDrive/estudo_etanol_gasolina_comum2022/precos-gasolina-etanol-08.csv",
                 sep=';')
df_set = pd.read_csv("/content/drive/MyDrive/estudo_etanol_gasolina_comum2022/precos-gasolina-etanol-09.csv",
                 encoding="iso-8859-1", sep=';')
df_out = pd.read_csv("/content/drive/MyDrive/estudo_etanol_gasolina_comum2022/precos-gasolina-etanol-10.csv",
                 sep=';')
df_nov = pd.read_csv("/content/drive/MyDrive/estudo_etanol_gasolina_comum2022/precos-gasolina-etanol-11.csv",
                 encoding="iso-8859-1", sep=';')
df_dez = pd.read_csv("/content/drive/MyDrive/estudo_etanol_gasolina_comum2022/precos-gasolina-etanol-12.csv",
                 sep=';')

### Organização dos dados
Os quadros de dados serão ordenados pela variável `Data da Coleta`, que contém a data em que o registro foi feito.

In [ ]:
frames = [df_jan, df_fev, df_mar, df_abr, df_mai, df_jun, df_jul, df_ago,
          df_set, df_out, df_nov, df_dez]

In [ ]:
frames_ord = [frame.sort_values(by="Data da Coleta") for frame in frames]

In [ ]:
df = pd.concat(frames_ord)
df

Regiao - Sigla Estado - Sigla    Municipio  \
0                  N             AC   RIO BRANCO   
1890              SE             MG     MANHUACU   
1891              SE             SP  SANTO ANDRE   
1892              SE             SP  SANTO ANDRE   
1893              SE             SP  SANTO ANDRE   
...              ...            ...          ...   
34797              S             SC     BLUMENAU   
34798              S             SC     BLUMENAU   
28962              S             SC     BLUMENAU   
28964              S             SC     BLUMENAU   
35379              S             SC     BLUMENAU   

                                  Revenda      CNPJ da Revenda  \
0               AUTO POSTO AMAPA - EIRELI   00.529.581/0001-53   
1890               AUTO POSTO ITAUNA LTDA   13.005.663/0001-40   
1891   POSTO DE GASOLINA MONTE CARLO LTDA   13.007.891/0001-50   
1892   POSTO DE GASOLINA MONTE CARLO LTDA   13.007.891/0001-50   
1893   POSTO DE GASOLINA MONTE CARLO LTDA   13.007.891/0001-50   
...                                   ...                  ...   
34797                AUTO POSTO M.L. LTDA   21.375.918/0001-95   
34798                AUTO POSTO M.L. LTDA   21.375.918/0001-95   
28962          AUTO POSTO BELA JOIA LTDA.   07.825.040/0001-11   
28964             AUTO POSTO BLU LIMITADA   07.134.578/0001-80   
35379              REDE DE POSTOS R4 LTDA   03.283.593/0006-43   

                Nome da Rua Numero Rua Complemento                   Bairro  \
0          VIA CHICO MENDES       3570         NaN                    AREAL   
1890     RUA LUIZ CERQUEIRA         55         NaN                   CENTRO   
1891      RUA JORGE BERETTA        960      TERREO  PARQUE ERASMO ASSSUNCAO   
1892      RUA JORGE BERETTA        960      TERREO  PARQUE ERASMO ASSSUNCAO   
1893      RUA JORGE BERETTA        960      TERREO  PARQUE ERASMO ASSSUNCAO   
...                     ...        ...         ...                      ...   
34797         RUA SAO PAULO       3149         NaN            ITOUPAVA SECA   
34798         RUA SAO PAULO       3149         NaN            ITOUPAVA SECA   
28962  RUA ANTONIO DA VEIGA        740      FRENTE            VICTOR KONDER   
28964  RUA SETE DE SETEMBRO       2093      FRENTE                   CENTRO   
35379       RUA DAS MISSOES       2426         NaN           NOVA ESPERANCA   

             Cep             Produto Data da Coleta Valor de Venda  \
0      69906-119            GASOLINA     03/01/2022           6,99   
1890   36900-085  GASOLINA ADITIVADA     03/01/2022          7,489   
1891   09271-400            GASOLINA     03/01/2022          6,397   
1892   09271-400              ETANOL     03/01/2022          4,997   
1893   09271-400  GASOLINA ADITIVADA     03/01/2022          6,597   
...          ...                 ...            ...            ...   
34797  89010-000  GASOLINA ADITIVADA     30/12/2022           5,32   
34798  89010-000            GASOLINA     30/12/2022           5,12   
28962  89012-500              ETANOL     30/12/2022           4,59   
28964  89012-401            GASOLINA     30/12/2022           4,49   
35379  89051-305  GASOLINA ADITIVADA     30/12/2022           5,31   

       Valor de Compra Unidade de Medida       Bandeira  
0                  NaN        R$ / litro  VIBRA ENERGIA  
1890               NaN        R$ / litro  VIBRA ENERGIA  
1891               NaN        R$ / litro       IPIRANGA  
1892               NaN        R$ / litro       IPIRANGA  
1893               NaN        R$ / litro       IPIRANGA  
...                ...               ...            ...  
34797              NaN        R$ / litro    RAIZEN MIME  
34798              NaN        R$ / litro    RAIZEN MIME  
28962              NaN        R$ / litro    RAIZEN MIME  
28964              NaN        R$ / litro       IPIRANGA  
35379              NaN        R$ / litro  VIBRA ENERGIA  

[603015 rows x 16 columns]

In [ ]:
df.shape

(603015, 16)

O dataframe `df` contém os dados de todo o Brasil, com um ottal de 603015 registros (linhas) e 16 variáveis (colunas).

### Populaçao de estudo
Será feita uma amostragem, contendo apenas os dados do estado do Amapá.

In [ ]:
df_ap = df.copy().rename(columns={"Estado - Sigla": "SIGLA_ESTADO"}).query("SIGLA_ESTADO == 'AP'")
df_ap.head()

Regiao - Sigla SIGLA_ESTADO Municipio  \
6664              N           AP    MACAPA   
6665              N           AP    MACAPA   
2882              N           AP    MACAPA   
2883              N           AP    MACAPA   
6741              N           AP    MACAPA   

                                  Revenda      CNPJ da Revenda  \
6664  AUTOMOTO - AUTOMOVEIS DO AMAPA LTDA   03.659.452/0013-01   
6665  AUTOMOTO - AUTOMOVEIS DO AMAPA LTDA   03.659.452/0015-73   
2882                     POSTO ICCAR LTDA   02.280.133/0034-52   
2883  AUTOMOTO COMBUSTIVEIS DO AMAPA LTDA   05.983.192/0003-89   
6741  AUTOMOTO - AUTOMOVEIS DO AMAPA LTDA   03.659.452/0006-82   

                      Nome da Rua Numero Rua            Complemento  \
6664   AVENIDA CARLOS LINS CORTES       1844                    NaN   
6665     AVENIDA HENRIQUE GALUCIO        161                LETRA A   
2882        AVENIDA RIO VILA NOVA         62  ESQ. C/ RUA BEIRA RIO   
2883           RUA CANDIDO MENDES        167                    NaN   
6741  RUA CICERO MARQUES DE SOUZA       3134                    NaN   

              Bairro        Cep   Produto Data da Coleta Valor de Venda  \
6664        INFRAERO  68908-074  GASOLINA     04/01/2022           5,86   
6665         CENTRAL  68900-115  GASOLINA     04/01/2022           5,86   
2882            TREM  68905-160  GASOLINA     04/01/2022           5,83   
2883         CENTRAL  68900-100  GASOLINA     04/01/2022           5,86   
6741  NOVO HORIZONTE  68909-803  GASOLINA     05/01/2022           5,86   

      Valor de Compra Unidade de Medida       Bandeira  
6664              NaN        R$ / litro         BRANCA  
6665              NaN        R$ / litro         BRANCA  
2882              NaN        R$ / litro         BRANCA  
2883              NaN        R$ / litro         BRANCA  
6741              NaN        R$ / litro  VIBRA ENERGIA

In [ ]:
df_ap.shape

(1391, 16)

O estado do Amapá possui um total de 1391 registros.

#### Renomeando variáveis
As variáveis serão colocadas em caixa-alta e separadas por underline.

In [ ]:
df_ap.rename(columns={"Regiao - Sigla": "SIGLA_REGIAO"},
             inplace=True)

In [ ]:
ajuste_variavel = {i:i.upper().replace(" ", "_") for i in df_ap.columns}

In [ ]:
df_ap.rename(columns=ajuste_variavel, inplace=True)

In [ ]:
df_ap.columns

Index(['SIGLA_REGIAO', 'SIGLA_ESTADO', 'MUNICIPIO', 'REVENDA',
       'CNPJ_DA_REVENDA', 'NOME_DA_RUA', 'NUMERO_RUA', 'COMPLEMENTO', 'BAIRRO',
       'CEP', 'PRODUTO', 'DATA_DA_COLETA', 'VALOR_DE_VENDA', 'VALOR_DE_COMPRA',
       'UNIDADE_DE_MEDIDA', 'BANDEIRA'],
      dtype='object')

#### Renomeando registros
Os registros do tipo object (texto) serão renomeados, para facilitar a visualização na tabela.

In [ ]:
registro_antigo = [[j for j in df_ap[i]] for i in df_ap.columns]

In [ ]:
ajuste_registro = [[j.title() if isinstance(j, str) else j
                    for j in df_ap[i]] for i in df_ap.columns]

In [ ]:
for var, i in zip(df_ap.columns, list(range(len(df_ap.columns)))):
    df_ap.replace(registro_antigo[i], ajuste_registro[i], inplace=True)

In [ ]:
df_ap.head()

SIGLA_REGIAO SIGLA_ESTADO MUNICIPIO                              REVENDA  \
6664            N           Ap    Macapa  Automoto - Automoveis Do Amapa Ltda   
6665            N           Ap    Macapa  Automoto - Automoveis Do Amapa Ltda   
2882            N           Ap    Macapa                     Posto Iccar Ltda   
2883            N           Ap    Macapa  Automoto Combustiveis Do Amapa Ltda   
6741            N           Ap    Macapa  Automoto - Automoveis Do Amapa Ltda   

          CNPJ_DA_REVENDA                  NOME_DA_RUA NUMERO_RUA  \
6664   03.659.452/0013-01   Avenida Carlos Lins Cortes       1844   
6665   03.659.452/0015-73     Avenida Henrique Galucio        161   
2882   02.280.133/0034-52        Avenida Rio Vila Nova         62   
2883   05.983.192/0003-89           Rua Candido Mendes        167   
6741   03.659.452/0006-82  Rua Cicero Marques De Souza       3134   

                COMPLEMENTO          BAIRRO        CEP   PRODUTO  \
6664                    NaN        Infraero  68908-074  Gasolina   
6665                Letra A         Central  68900-115  Gasolina   
2882  Esq. C/ Rua Beira Rio            Trem  68905-160  Gasolina   
2883                    NaN         Central  68900-100  Gasolina   
6741                    NaN  Novo Horizonte  68909-803  Gasolina   

     DATA_DA_COLETA VALOR_DE_VENDA  VALOR_DE_COMPRA UNIDADE_DE_MEDIDA  \
6664     04/01/2022           5,86              NaN        R$ / Litro   
6665     04/01/2022           5,86              NaN        R$ / Litro   
2882     04/01/2022           5,83              NaN        R$ / Litro   
2883     04/01/2022           5,86              NaN        R$ / Litro   
6741     05/01/2022           5,86              NaN        R$ / Litro   

           BANDEIRA  
6664         Branca  
6665         Branca  
2882         Branca  
2883         Branca  
6741  Vibra Energia

In [ ]:
df_ap.dtypes

SIGLA_REGIAO          object
SIGLA_ESTADO          object
MUNICIPIO             object
REVENDA               object
CNPJ_DA_REVENDA       object
NOME_DA_RUA           object
NUMERO_RUA            object
COMPLEMENTO           object
BAIRRO                object
CEP                   object
PRODUTO               object
DATA_DA_COLETA        object
VALOR_DE_VENDA        object
VALOR_DE_COMPRA      float64
UNIDADE_DE_MEDIDA     object
BANDEIRA              object
dtype: object

#### Estratificação da população
A população será estratificada com base na gasolina do tipo comum e na revenda com maior ocorrência de dados.

In [ ]:
df_ap.PRODUTO.value_counts()

Gasolina              1163
Gasolina Aditivada     183
Etanol                  45
Name: PRODUTO, dtype: int64

In [ ]:
df_ap.REVENDA.value_counts()

Automoto - Automoveis Do Amapa Ltda          331
Automoto Combustiveis Do Amapa Ltda          193
Sepe Tiaraju Empreendimentos Eireli          115
Comercial Grao De Ouro Ltda                  106
Posto De Combustiveis Jardins Eireli         102
Claudionor Costa Dos Santos                   83
Posto Iccar Ltda                              70
Grampos Oiapoc Ltda                           64
Real Petróleo Ltda                            63
J C Da S Farias Eireli                        62
Auto Posto Machado Ltda                       62
Auto Posto Terceiro Milênio Ltda              38
Souza & Cavalcante Comercio Ltda              24
Monte & Filhos Ltda                           21
Aeroposto Combustiveis Automotivos Ltda       18
J. C. S. Guimaraes Ltda                       14
Posto Catarinao Eireli                        10
Auto Posto Playcenter Ltda                     4
Machado & Andrade Ltda                         3
Auto Posto Amazonas Stn Ltda                   2
R E R Empreendimento

A revenda com maior ocorrência de dados é a Automoto - Automoveis Do Amapa Ltda.

In [ ]:
automoto_atmv = df_ap.copy().query("PRODUTO == 'Gasolina'\
and REVENDA == 'Automoto - Automoveis Do Amapa Ltda'")
automoto_atmv.head()

SIGLA_REGIAO SIGLA_ESTADO MUNICIPIO  \
6664             N           Ap    Macapa   
6665             N           Ap    Macapa   
6741             N           Ap    Macapa   
6743             N           Ap    Macapa   
12180            N           Ap    Macapa   

                                   REVENDA      CNPJ_DA_REVENDA  \
6664   Automoto - Automoveis Do Amapa Ltda   03.659.452/0013-01   
6665   Automoto - Automoveis Do Amapa Ltda   03.659.452/0015-73   
6741   Automoto - Automoveis Do Amapa Ltda   03.659.452/0006-82   
6743   Automoto - Automoveis Do Amapa Ltda   03.659.452/0002-59   
12180  Automoto - Automoveis Do Amapa Ltda   03.659.452/0014-92   

                       NOME_DA_RUA NUMERO_RUA COMPLEMENTO          BAIRRO  \
6664    Avenida Carlos Lins Cortes       1844         NaN        Infraero   
6665      Avenida Henrique Galucio        161     Letra A         Central   
6741   Rua Cicero Marques De Souza       3134         NaN  Novo Horizonte   
6743             Rua Hildemar Maia        900           A      Santa Rita   
12180             Rua Eliezer Levy       1047     Letra A         Central   

             CEP   PRODUTO DATA_DA_COLETA VALOR_DE_VENDA  VALOR_DE_COMPRA  \
6664   68908-074  Gasolina     04/01/2022           5,86              NaN   
6665   68900-115  Gasolina     04/01/2022           5,86              NaN   
6741   68909-803  Gasolina     05/01/2022           5,86              NaN   
6743   68901-310  Gasolina     05/01/2022           5,86              NaN   
12180  68900-083  Gasolina     06/01/2022           5,86              NaN   

      UNIDADE_DE_MEDIDA       BANDEIRA  
6664         R$ / Litro         Branca  
6665         R$ / Litro         Branca  
6741         R$ / Litro  Vibra Energia  
6743         R$ / Litro         Branca  
12180        R$ / Litro         Branca

Verificando se há registros vazios na variável VALOR_DE_VENDA.

In [ ]:
automoto_atmv.VALOR_DE_VENDA.isnull().value_counts()

False    330
Name: VALOR_DE_VENDA, dtype: int64

Serão removidas variáveis que não serão utilizadas nos estudos.

In [ ]:
automoto_atmv.SIGLA_REGIAO.value_counts()

N    330
Name: SIGLA_REGIAO, dtype: int64

In [ ]:
automoto_atmv.SIGLA_ESTADO.value_counts()

Ap    330
Name: SIGLA_ESTADO, dtype: int64

In [ ]:
automoto_atmv.VALOR_DE_COMPRA.isnull().value_counts()

True    330
Name: VALOR_DE_COMPRA, dtype: int64

In [ ]:
automoto_atmv.drop(columns=["PRODUTO", "REVENDA", "SIGLA_ESTADO",
                            "SIGLA_REGIAO", "VALOR_DE_COMPRA"], inplace=True)

In [ ]:
automoto_atmv.shape

(330, 11)

Será necessário fazer uma mudança do tipo da variável `VALOR_DE_VENDA`, suprindo a necessidade de operações numéricas do estudo.

In [ ]:
vv_novo = [i.replace(",", ".") for i in automoto_atmv.VALOR_DE_VENDA]

In [ ]:
automoto_atmv.VALOR_DE_VENDA.replace(automoto_atmv.VALOR_DE_VENDA.values,
                                     vv_novo, inplace=True)

In [ ]:
automoto_atmv["VALOR_DE_VENDA"] = automoto_atmv.VALOR_DE_VENDA.astype(float)

A população a ser utilizada no estudo contemplará os 4 bairros com maior quantidade de dados.

In [ ]:
automoto_atmv.BAIRRO.value_counts()

Central           103
Pacoval            47
Santa Rita         40
Buritizal          39
Trem               28
Novo Horizonte     22
Infraero           21
Congos             12
Novo Buritizal     11
Sao Jose            7
Name: BAIRRO, dtype: int64

In [ ]:
bairros = list(automoto_atmv.BAIRRO.value_counts().index)[:4]

In [ ]:
automoto_atmv2 = automoto_atmv.copy()[automoto_atmv["BAIRRO"].isin(bairros)]
automoto_atmv2.head()

MUNICIPIO      CNPJ_DA_REVENDA               NOME_DA_RUA NUMERO_RUA  \
6665     Macapa   03.659.452/0015-73  Avenida Henrique Galucio        161   
6743     Macapa   03.659.452/0002-59         Rua Hildemar Maia        900   
12180    Macapa   03.659.452/0014-92          Rua Eliezer Levy       1047   
11802    Macapa   03.659.452/0007-63         Rua Hildemar Maia       2699   
11100    Macapa   03.659.452/0001-78             Rua Guanabara        503   

      COMPLEMENTO      BAIRRO        CEP DATA_DA_COLETA  VALOR_DE_VENDA  \
6665      Letra A     Central  68900-115     04/01/2022            5.86   
6743            A  Santa Rita  68901-310     05/01/2022            5.86   
12180     Letra A     Central  68900-083     06/01/2022            5.86   
11802         NaN   Buritizal  68902-870     06/01/2022            5.86   
11100         NaN     Pacoval  68908-360     06/01/2022            5.86   

      UNIDADE_DE_MEDIDA BANDEIRA  
6665         R$ / Litro   Branca  
6743         R$ / Litro   Branca  
12180        R$ / Litro   Branca  
11802        R$ / Litro   Branca  
11100        R$ / Litro   Branca

# DESCRIÇÃO DOS DADOS

## Medidas de Tendência Central
Com as Medidas de Tendência Central, o conjunto de dados pode ser representado como um todo, com suas características sendo identificadas.

Tendo em vista que o conjunto de dados é datado ao ano de 2022, será feito um ajuste na data, para facilitar a visualização dos dados.

In [ ]:
ajuste_data = [i.replace("/2022", '') for i in automoto_atmv2.DATA_DA_COLETA]

In [ ]:
automoto_atmv2.DATA_DA_COLETA.replace(automoto_atmv2.DATA_DA_COLETA.values,
                                      ajuste_data, inplace=True)

Serão criado 4 subdataframes, para as medidas de tendência central e uso posterior.

In [ ]:
atmv_bairro1 = automoto_atmv2[automoto_atmv2.BAIRRO == bairros[0]]
atmv_bairro2 = automoto_atmv2[automoto_atmv2.BAIRRO == bairros[1]]
atmv_bairro3 = automoto_atmv2[automoto_atmv2.BAIRRO == bairros[2]]
atmv_bairro4 = automoto_atmv2[automoto_atmv2.BAIRRO == bairros[3]]

**Média:**  
Com a média, têm-se a concentração dos dados.

In [ ]:
md_b1 = round(atmv_bairro1.VALOR_DE_VENDA.mean(), 3)
md_b2 = round(atmv_bairro2.VALOR_DE_VENDA.mean(), 3)
md_b3 = round(atmv_bairro3.VALOR_DE_VENDA.mean(), 3)
md_b4 = round(atmv_bairro4.VALOR_DE_VENDA.mean(), 3)

**Moda:**  
A moda mostra o valor, ou valores que mais se repetem nos dados.

In [ ]:
mod_b1 = list(atmv_bairro1.VALOR_DE_VENDA.mode())
mod_b2 = list(atmv_bairro2.VALOR_DE_VENDA.mode())
mod_b3 = list(atmv_bairro3.VALOR_DE_VENDA.mode())
mod_b4 = list(atmv_bairro4.VALOR_DE_VENDA.mode())

**Mediana:**  
Com a mediana obtemos o valor central no dataset.

In [ ]:
mdn_b1 = round(atmv_bairro1.VALOR_DE_VENDA.median(), 3)
mdn_b2 = round(atmv_bairro2.VALOR_DE_VENDA.median(), 3)
mdn_b3 = round(atmv_bairro3.VALOR_DE_VENDA.median(), 3)
mdn_b4 = round(atmv_bairro4.VALOR_DE_VENDA.median(), 3)

In [ ]:
dados_mtc = {"MEDIA": [md_b1, md_b2, md_b3, md_b4],
             "MODA": [mod_b1, mod_b2, mod_b3, mod_b4],
             "MEDIANA": [mdn_b1, mdn_b2, mdn_b3, mdn_b4]}

In [ ]:
df_mtc = pd.DataFrame(dados_mtc, index=bairros)
df_mtc

MEDIA          MODA  MEDIANA
Central     5.486        [6.44]    5.790
Pacoval     5.504  [6.25, 6.44]    5.790
Santa Rita  5.463        [6.44]    5.775
Buritizal   5.549  [6.25, 6.44]    5.820

Apesar de se tratar de 4 postos de vendas diferentes, as medidas de tendência central encontram-se em coesão, pois tratam-se da mesma revenda.

## Visualização dos Dados

### Boxplot
O boxplot representa a distribuição dos dados no conjunto. O mesmo contém os 4 postos com maior quantidade de dados.

In [ ]:
import plotly.express as px

In [ ]:
tema = px.colors.qualitative.Light24

In [ ]:
boxplot_precos = px.box(automoto_atmv2, y="VALOR_DE_VENDA", color="BAIRRO",
                   points="all", notched=True,
                   color_discrete_sequence=tema)

boxplot_precos.update_layout(width=1000, plot_bgcolor="#F9F9F9",
                        title=dict(text="Automoto - Automoveis Do Amapá\
 Ltda.<br>Preço da Gasolina 2022<br>", x=0.5, y=0.96, font_size=22),
                        legend=dict(title="Bairro", bgcolor="#F9F9F9",
                                    bordercolor="#D0D1D3", borderwidth=1,
                                    font=dict(size=14)))

boxplot_precos.update_xaxes(showline=True, linewidth=1, linecolor="lightgrey")

boxplot_precos.update_yaxes(title="Valor de Venda", showline=True, linewidth=1,
                            linecolor="lightgrey", tickprefix="R$")

Com este resultado, é possível perceber que a amplitude escalar do preço da gasolina entre os 4 postos encontra semelhança entre si, bem como a centralidade dos dados nos conjuntos. É possível notar ainda que, entre a mediana e o limite inferior dos dados, há uma dispersão maior no conjunto de dados, enquanto que entre a mediana e o limite superior, os dados estão mais concentrados.

### Escala de preços
A escala de preços em relação a data revela a variação do preço da gasolina comum em determinada data.

In [ ]:
from plotly.subplots import make_subplots
import plotly.graph_objects as go

In [ ]:
dados_precos = go.Scatter(x=automoto_atmv2.DATA_DA_COLETA,
                          y=automoto_atmv2.VALOR_DE_VENDA, mode="lines",
                          line=dict(color=tema[0]))

layout_precos = go.Layout(title=dict(text="Escala de Preços 2022 - \
Postos Automoto", x=0.5, font_size=22),
                          xaxis=dict(title="Data"),
                          yaxis=dict(title="Preço"), height=600)

indice_precos = go.Figure(data=dados_precos, layout=layout_precos)

indice_precos.update_layout(xaxis_range=[15, 50], plot_bgcolor="#F9F9F9")

indice_precos.update_yaxes(showline=True, linewidth=1, linecolor="lightgrey",
                           tickprefix="R$")
indice_precos.update_xaxes(showline=True, linewidth=1, linecolor="lightgrey",
                           rangeslider_visible=True)

Com base na representação gráfica, é possível notar que entre o período de 25/04 a 22/06, houve uma consistência no preço da gasolina. Após este período,houve uma decrescente até o dia 28/09.  
Vejamos esta relação, tendo em vista os 4 postos analisados anteriormente, de forma separada.

In [ ]:
quadro_precos = make_subplots(rows=2, cols=2, subplot_titles=bairros)

quadro_precos.add_trace(go.Scatter(x=atmv_bairro1.DATA_DA_COLETA,
                         y=atmv_bairro1.VALOR_DE_VENDA,
                         line=dict(color=tema[0]),
                         name=bairros[0]), row=1, col=1)

quadro_precos.add_trace(go.Scatter(x=atmv_bairro2.DATA_DA_COLETA,
                         y=atmv_bairro2.VALOR_DE_VENDA,
                         line=dict(color=tema[1]),
                         name=bairros[1]), row=1, col=2)

quadro_precos.add_trace(go.Scatter(x=atmv_bairro3.DATA_DA_COLETA,
                         y=atmv_bairro3.VALOR_DE_VENDA,
                         line=dict(color=tema[2]),
                         name=bairros[2]), row=2, col=1)

quadro_precos.add_trace(go.Scatter(x=atmv_bairro4.DATA_DA_COLETA,
                         y=atmv_bairro4.VALOR_DE_VENDA,
                         line=dict(color=tema[3]),
                         name=bairros[3]), row=2, col=2)

quadro_precos.update_traces(mode="lines")

quadro_precos.update_layout(height=700, plot_bgcolor="white",
                  title=dict(text="Automoto - Automoveis Do Amapá Ltda.<br>\
Preço da Gasolina 2022<br>", x=0.5, y=0.95, font_size=22),
                  legend=dict(title="Bairros", bgcolor="#F9F9F9",
                              bordercolor="#D0D1D3", borderwidth=1,
                              font=dict(size=14)))

quadro_precos.update_xaxes(showline=True, linewidth=1, linecolor="lightgrey",
                 tickangle=45)
quadro_precos.update_yaxes(showline=True, linewidth=1, linecolor="lightgrey",
                 tickprefix="R$")

Há similaridade quanto a visualização gráfica em relação ao preço da gasolina comum entre os 4 postos.

# SALVANDO O CONJUNTO DE DADOS
É possível salvar o conjunto de dados tratado contendo todos os postos de vendas dos postos Automoto Automóveis do Amapá Ltda. em um arquivo csv.

In [ ]:
automoto_atmv.to_csv("gasolina_2022_automoto_atmv.csv", index=False)